src/NativeLog/Log/Log.h

```diff
@@ -304,7 +304,7 @@ class Log final : private boost::noncopyable
 
     std::atomic_flag closed;
 
-    bool inmemory = false;
+    std::atomic<bool> inmemory = false;
```

Comment:
Why this need to be atomic since it is set once during ctor and read-only afterwards ?

Reply:
the boolean var also report by thread sanitizer,  basically the usage(read) does not been guarantee the thread safety.

Reply:
Interesting. Can we mask this variable to avoid the TSA ?

Reply:
https://github.com/timeplus-io/proton/issues/320#issuecomment-1842414990